app/src/main/java/org/xbmc/kore/ui/sections/settings/SettingsFragment.java

```diff
@@ -217,4 +233,76 @@ public boolean onPreferenceClick(Preference preference) {
             }
         });
     }
+
+    private void setupLanguagePreference(final ListPreference languagePref) {
+        Locale[] locales = getLocales(BuildConfig.SUPPORTED_LOCALES);
+
+        Arrays.sort(locales, new Comparator<Locale>() {
+            @Override
+            public int compare(Locale o1, Locale o2) {
+                return o1.getDisplayName().compareToIgnoreCase(o2.getDisplayName());
+            }
+        });
+
+        String[] displayNames = new String[locales.length];
+        String[] entryValues = new String[locales.length];
+        for(int index = 0; index < locales.length; index++) {
+            Locale locale = locales[index];
+            displayNames[index] = locale.getDisplayName(locale);
```

Comment:
Should be `getDisplayName(currentLocale);`

Reply:
I don't agree. If you change the language, either by mistake or just to try it out, it can be really hard to change it back if you don't understand the language at all.

Reply:
Ok